src/context/data.tsx

```diff
@@ -83,36 +80,52 @@ export const DataProvider: React.FC<React.PropsWithChildren> = ({
 
   // create state for synchronization
   const [synced, setSynced] = useState<boolean>(false);
+
   //TODO: Use real values instread.
-  const [filterValues, setFilterValues] = useState<FilterValues>({
-    radius: { lower: 4, upper: 50 },
-    height: { lower: 3, upper: 43 },
-  });
+  const [filterValues, setCurrentFilterValues] = useState<FilterValues | undefined>(undefined);
 
   // use the offline context
   const { inventory } = useOffline();
 
+  // define the setFilterValues function
+  const setFilterValues = (newValues: FilterValues) => {
+    setCurrentFilterValues({
+      height: { ...newValues.height },
+      radius: { ...newValues.radius}
+    })
+  } 
+
```

Comment:
You need to wrap the state update handler function by a function, that will leave the context.